src/AutoRest.CSharp/Common/Output/Models/Types/ClientOptionsTypeProvider.cs

```diff
@@ -16,6 +17,7 @@ internal sealed class ClientOptionsTypeProvider : TypeProvider
 
         public FormattableString Description { get; }
         public IReadOnlyList<ApiVersion>? ApiVersions { get; }
+        public IReadOnlyList<Parameter>? AdditionalProperties { get; init; }
```

Comment:
looks like in your code, we will always assign an instance to this property.
Also it really does not have a difference when it is null and it is an empty list - it represents there is no additional properties.
Therefore we could make it `IReadOnlyList<Parameter>` (remove the question mark)

Reply:
> we will always assign an instance to this property.

No. Only DPG will do that, HLC not.

Reply:
Why not just return empty collection, so that we can remove `?` here? If `null` doesn't have extra meaning here, I'd like to recommend to remove `?`.

Reply:
empty collection here does not change anything for HLC

Reply:
If this is the issue:

> if we remove the question mark, we will have to assign it in the ctor and in HLC it does not have anything to assign

then we should
1. add a new ctor that accepts an `IReadOnlyList<Parameter>` to assign value to this property
2. the old ctor calls `this` with `Array.Empty<Parameter>` for this property

in this way, HLC is unchanged as well